tests/components/lock/test_init.py

```diff
@@ -2,328 +2,261 @@
+    SERVICE_OPEN,
+    SERVICE_UNLOCK,
     STATE_JAMMED,
     STATE_LOCKED,
     STATE_LOCKING,
     STATE_UNLOCKED,
     STATE_UNLOCKING,
-    LockEntity,
     LockEntityFeature,
 )
 from homeassistant.core import HomeAssistant
 import homeassistant.helpers.entity_registry as er
-from homeassistant.setup import async_setup_component
+from homeassistant.helpers.typing import UNDEFINED, UndefinedType
 
 from tests.testing_config.custom_components.test.lock import MockLock
 
+TEST_ENTITY_ID = "lock.test_lock"
 
-class MockLockEntity(LockEntity):
-    """Mock lock to use in tests."""
-
-    def __init__(
-        self,
-        code_format: str | None = None,
-        lock_option_default_code: str = "",
-        supported_features: LockEntityFeature = LockEntityFeature(0),
-    ) -> None:
-        """Initialize mock lock entity."""
-        self._attr_supported_features = supported_features
-        self.calls_lock = MagicMock()
-        self.calls_unlock = MagicMock()
-        self.calls_open = MagicMock()
-        if code_format is not None:
-            self._attr_code_format = code_format
-        self._lock_option_default_code = lock_option_default_code
-
-    async def async_lock(self, **kwargs: Any) -> None:
-        """Lock the lock."""
-        self.calls_lock(kwargs)
-        self._attr_is_locking = False
-        self._attr_is_locked = True
-
-    async def async_unlock(self, **kwargs: Any) -> None:
-        """Unlock the lock."""
-        self.calls_unlock(kwargs)
-        self._attr_is_unlocking = False
-        self._attr_is_locked = False
-
-    async def async_open(self, **kwargs: Any) -> None:
-        """Open the door latch."""
-        self.calls_open(kwargs)
-
-
-async def test_lock_default(hass: HomeAssistant) -> None:
-    """Test lock entity with defaults."""
-    lock = MockLockEntity()
-    lock.hass = hass
 
-    assert lock.code_format is None
-    assert lock.state is None
+async def help_test_async_lock_service(
+    hass: HomeAssistant, service: str, code: str | None | UndefinedType = UNDEFINED
+) -> None:
+    """Help to lock a test lock."""
+    data: dict[str, Any] = {"entity_id": TEST_ENTITY_ID}
+    if code is not UNDEFINED:
+        data[ATTR_CODE] = code
 
+    await hass.services.async_call(DOMAIN, service, data, blocking=True)
 
-async def test_lock_states(hass: HomeAssistant) -> None:
-    """Test lock entity states."""
 
-    lock = MockLockEntity()
-    lock.hass = hass
+async def test_lock_default(hass: HomeAssistant, mock_lock: MockLock) -> None:
+    """Test lock entity with defaults."""
 
-    assert lock.state is None
+    assert mock_lock.code_format is None
+    assert mock_lock.state is None
 
-    lock._attr_is_locking = True
-    assert lock.is_locking
-    assert lock.state == STATE_LOCKING
 
-    await lock.async_handle_lock_service()
-    assert lock.is_locked
-    assert lock.state == STATE_LOCKED
+async def test_lock_states(hass: HomeAssistant, mock_lock: MockLock) -> None:
```

Comment:
There seem to be 3 tests in one test function: locking, unlocking, jammed. Would it be a good idea to have 3 separate tests? This is usually a good practice in unit testing. (my 2 p)

Reply:
May be, but in this PR the aim is just to move the tests to use the entity platform.
Personally, I see not an issue in this as this is about testing lock states.

Reply:
I see that this PR is mainly for something else than changing the tests (namely use the entity platform), so you keep change to the minimum. Understood.